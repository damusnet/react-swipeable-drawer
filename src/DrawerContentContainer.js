import React from "react";
import PropTypes from "prop-types";

const DrawerContentContainer = ({
  width,
  swiping,
  translateX,
  handleTouchStart,
  handleTouchMove,
  handleTouchEnd,
  drawerContent,
}) => (
  <div
    className="DrawerContentContainer"
    onTouchStart={handleTouchStart}
    onTouchMove={handleTouchMove(width)}
    onTouchEnd={handleTouchEnd}
    style={{
      position: "fixed",
      top: 0,
      bottom: 0,
      zIndex: 1,
      left: `-${width}%`,
      width: `${width}%`,
      transform: `translateX(${translateX}%)`,
      transition: swiping ? "" : "transform .2s ease-in-out",
    }}
  >
    {drawerContent}
  </div>
);

export default DrawerContentContainer;

DrawerContentContainer.propTypes = {
  width: PropTypes.number.isRequired,
  swiping: PropTypes.bool.isRequired,
  translateX: PropTypes.number.isRequired,
  handleTouchStart: PropTypes.func.isRequired,
  handleTouchMove: PropTypes.func.isRequired,
  handleTouchEnd: PropTypes.func.isRequired,
  drawerContent: PropTypes.element.isRequired,
};
