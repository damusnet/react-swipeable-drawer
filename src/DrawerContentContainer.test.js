import React from "react";
import renderer from "react-test-renderer";

import DrawerContentContainer from "./DrawerContentContainer";

const noop = () => () => {};

describe("<DrawerContentContainer />", () => {
  it("renders correctly when closed", () => {
    const component = renderer.create(
      <DrawerContentContainer
        position="left"
        size={80}
        zIndex={2}
        swiping={false}
        translation={0}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when swiping right", () => {
    const component = renderer.create(
      <DrawerContentContainer
        position="left"
        size={80}
        zIndex={2}
        swiping
        translation={50}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when swiping left", () => {
    const component = renderer.create(
      <DrawerContentContainer
        position="right"
        size={80}
        zIndex={2}
        swiping
        translation={50}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when swiping down", () => {
    const component = renderer.create(
      <DrawerContentContainer
        position="top"
        size={80}
        zIndex={2}
        swiping
        translation={50}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when up", () => {
    const component = renderer.create(
      <DrawerContentContainer
        position="bottom"
        size={80}
        zIndex={2}
        swiping
        translation={50}
        handleTouchStart={noop}
        handleTouchMove={noop}
        handleTouchEnd={noop}
        drawerContent={<div />}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
