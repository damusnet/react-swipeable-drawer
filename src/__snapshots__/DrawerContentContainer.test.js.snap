// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerContentContainer /> renders correctly when closed 1`] = `
<div
  className="DrawerContentContainer"
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": "-80%",
      "position": "fixed",
      "top": 0,
      "transform": "translateX(0%)",
      "transition": "transform .2s ease-in-out",
      "width": "80%",
      "zIndex": 1,
    }
  }
>
  <div />
</div>
`;

exports[`<DrawerContentContainer /> renders correctly when swiping 1`] = `
<div
  className="DrawerContentContainer"
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": "-80%",
      "position": "fixed",
      "top": 0,
      "transform": "translateX(50%)",
      "transition": "",
      "width": "80%",
      "zIndex": 1,
    }
  }
>
  <div />
</div>
`;
